testing/board/test_board_vc.cpp

```diff
@@ -0,0 +1,358 @@
+* @brief      Tests: read_phy_group_info
+* @details    Validates fpga phy group information  <br>
+*/
+TEST_P(board_vc_c_p, board_vc_6) {
+
+	int group_num = 0;
+
+	EXPECT_EQ(read_phy_group_info(tokens_[0], NULL, &group_num), FPGA_OK);
+
+	EXPECT_NE(read_phy_group_info(tokens_[0], NULL, NULL), FPGA_OK);
+
+	EXPECT_NE(read_phy_group_info(NULL, NULL, &group_num), FPGA_OK);
+}
+
+/**
+* @test       board_vc_7
+* @brief      Tests: print_board_info
+* @details    Validates fpga board info  <br>
+*/
+TEST_P(board_vc_c_p, board_vc_7) {
+
+	EXPECT_EQ(print_board_info(tokens_[0]), FPGA_OK);
+
+	EXPECT_EQ(print_mac_info(tokens_[0]), FPGA_OK);
+
+	EXPECT_EQ(print_phy_info(tokens_[0]), FPGA_OK);
+}
+
+/**
+* @test       board_vc_8
+* @brief      Tests: read_max10fw_version
+*             read_bmcfw_version,
+* @details    Validates fpga invalid fpga firmware version  <br>
+*/
+TEST_P(board_vc_c_p, board_vc_8) {
+
+	char buf[10] = { 0 };
+	write_sysfs_file((const char *)"spi-altera.0.auto/spi_master/spi0/spi0.0/bmcfw_flash_ctrl/bmcfw_version", (void*)buf, sizeof(buf));
+
+	char bmcfw_ver[SYSFS_MAX_SIZE];
+	EXPECT_NE(read_bmcfw_version(tokens_[0], bmcfw_ver, SYSFS_MAX_SIZE), FPGA_OK);
+
+	write_sysfs_file((const char *)"spi-altera.0.auto/spi_master/spi0/spi0.0/max10_version", (void*)buf, sizeof(buf));
+
+	char max10fw_ver[SYSFS_MAX_SIZE];
+	EXPECT_NE(read_max10fw_version(tokens_[0], max10fw_ver, SYSFS_MAX_SIZE), FPGA_OK);
+}
+INSTANTIATE_TEST_CASE_P(baord_vc_c, board_vc_c_p,
+	::testing::ValuesIn(test_platform::mock_platforms({ "dcp-vc" })));
+
+// test invalid sysfs attributes
+class board_vc_invalid_c_p : public board_vc_c_p { };
+
+/**
+* @test       board_vc_9
+* @brief      Tests: read_max10fw_version
+*             read_max10fw_version,read_pcb_info
+*             read_pkvl_info,read_mac_info
+*             read_phy_group_info,print_board_info
+*             print_phy_info,print_mac_info
+* @details    Validates function with invalid sysfs <br>
+*/
+TEST_P(board_vc_invalid_c_p, board_vc_9) {
+
+	char bmcfw_ver[SYSFS_MAX_SIZE];
+	EXPECT_NE(read_bmcfw_version(tokens_[0], bmcfw_ver, SYSFS_MAX_SIZE), FPGA_OK);
+
+	char max10fw_ver[SYSFS_MAX_SIZE];
+	EXPECT_NE(read_max10fw_version(tokens_[0], max10fw_ver, SYSFS_MAX_SIZE), FPGA_OK);
+
+	char pcb_info[SYSFS_MAX_SIZE];
+	EXPECT_NE(read_pcb_info(tokens_[0], pcb_info, SYSFS_MAX_SIZE), FPGA_OK);
+
+	fpga_pkvl_info pkvl_info;
+	int fpga_mode;
+	EXPECT_NE(read_pkvl_info(tokens_[0], &pkvl_info, &fpga_mode), FPGA_OK);
+
+	unsigned char buf[8] = { 0 };
+	EXPECT_NE(read_mac_info(tokens_[0], buf, 8), FPGA_OK);
+
+	int group_num = 0;
+	EXPECT_NE(read_phy_group_info(tokens_[0], NULL, &group_num), FPGA_OK);
+
+	EXPECT_NE(print_board_info(tokens_[0]), FPGA_OK);
+
+	EXPECT_NE(print_mac_info(tokens_[0]), FPGA_OK);
+
+	EXPECT_NE(print_phy_info(tokens_[0]), FPGA_OK);
+}
+INSTANTIATE_TEST_CASE_P(board_vc_invalid_c, board_vc_invalid_c_p,
+	::testing::ValuesIn(test_platform::mock_platforms({ "skx-p" })));
```

Comment:
skx-p is probably not the one we want to use for these tests.

Reply:
negative test cases

Reply:
fixed